docs/index.md

```diff
@@ -0,0 +1,187 @@
+  if city == "france-paris" OR city == "france-lyon" then 4.0
+  if city == "italy-milan" then 3.0
+  if true then 1.0
+```
+
+## Creating and updating rulesets
+
+Stakeholders can agree with developers and they will create a ruleset. Once created, stakeholders can edit rulesets in full autonomy, with the guarantee of not breaking the code. Updating a ruleset means, in concrete terms to either change one or multiple rules, by changing their conditions or their result. 
+
+Whether to use Regula or not for a given business problem should be done by answering the following questions:
+
+- the behaviour is changing often or is supposed to evolve a lot
+- the behaviour is relying entirely on business facts 
+
+If the answer is yes for both of them, then creating a ruleset is a good idea. 
+
+
+In the current release of Regula (`0.5.0`) stakeholders can only create or edits rules through scripts which require some engineering knowledge. Later version will include a proper UI for stakeholders to interact with Regula in a simple manner. 
+
+
+## Components overview 
+
+> ⚠ TODO: we should discuss what we want to bring in here, this can get quite complex really quick.
+
+![Cinematic](./cinematic.png)
+
+- Regula clients are querying Regula server
+  - they live in the source code of a given application
+  - when high performances are required, i.e. ruleset evaluations should be as fast as possible, there are caching mechanism available in some clients. See more about this in the technical documentation.
+  
+- Regula server answers requests from the clients. 
+- Rulesets storage is where the rulesets and their versions are being kept. It is not supposed to be interacted with directly.
+
+## Design principles
+
+Rules engines are delicate beasts. It is important to remember that technically, it's about extracting behaviour from the source code and relocating it outside in a more "accessible" place that is outside of any continuous integration mechanism. 
+
+Regula takes an opinionated approach to lower the risks and make it harder to break from expected behaviours. Regula's team objective is to focus on a curated feature set, putting reliability and predictability at a top priority. 
+
+## Predictability 
+
+Regula, by splitting the ownership of a given functionality of the software that uses it, needs to be predictable. Behaviour being owned by whoever edits the rules, it is mandatory that a given ruleset is not able to break the software that uses it. 
+
+On developers side:
+
+Regula is "typed" and that is why the concept of **signature** exists, illustrated by:
+
+```
+float64 /marketplace/radius(string city, string driver-status)
+```
+
+By doing so, it allows developers of both typed and dynamic languages to be certain that they always get a value of a given type. Changing the signature implies to create a new rule and such a break will logically require to edit the source code of the application using Regula to reflect the change. 
```

Comment:
"implies to create" should be "implies the creation of" and "requires to edit" should be "requires modifying".